torchgeo/transforms/transforms.py

```diff
@@ -73,3 +73,129 @@ def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
             sample["boxes"] = sample["boxes"].to(boxes_dtype)
 
         return sample
+
+
+class ConstantNormalize(Module):
```

Comment:
- K.Normalize does (x-mean)/std normalization. I saw that we have a lot of datamodules for non-geodatasets with rgb imagery where we just do scaling to 0-1 by dividing by 255. I guess you could also use K.Normalize(mean=0, std=255) to do the same but I thought a ConstantNormalize is a bit less hacky?
- K.RandomCrop will only crop a single image once, but we want to create n patches from a single image so `RepeatedRandomCrop` is a sort of wrapper around `K.RandomCrop` that does this patchifying 
- from what I understand K.PadTo would have to be initialized with a desired `h,w` once before passing it to the transform, however, we have several datasets where the images are not of constant size but actually vary quiet a bit, so for each sample we have to compute the next available multiple of 32 in this case and pad according to that

Reply:
* I think `K.Normalize` would be better since we don't need to maintain any additional code in TorchGeo
* torchvision has a 5-crop and 10-crop, I think we could argue for having an N-crop. Would prob be easiest to implement this in TorchGeo for now and try to contribute it to Kornia so that we can remove it someday
* Can we not just choose a consistent size and always pad to that? How widely do these sizes vary?

Reply:
- okay, will change that
- okay, will change the name to N-crop? or would NCrop be different from the current `RepeatedRandomCrop`
- looking at the datasets doc page: vaihingen (1281-3816), SpaceNet (102-900), OSCD (40-1180) are the ones with quiet a bit of variation

Reply:
* :+1:
* I like `NCrop` but we'll see what the Kornia devs think. Can open a PR simultaneous to this one to start getting feedback
* But we're cropping all of those first, right? Won't the crops have approximately consistent dimensions?

Reply:
- last point: no when you are cropping you are choosing a patch_size that should be a multiple of 32, the dynamic padding is done for eval/test samples that come in the full tile size and are adjusted to fit the model architecture

Reply:
Maybe we can upstream a PadToMultiple? Not sure if that form of padding would be useful outside of satellite imagery. Curious what @isaaccorley thinks (I think @calebrob6 is officially on vacation now)

Reply:
It seems like a `PadToMultiple` should be useful for segmentation tasks, however, I think most computer vision segmentation datasets are curated so that its often already taken care of.

Reply:
I think there might be another issue:

When using `K.Normalize` I need to wrap it in order for the this single augmentation to work with our dictionaries. If I use the torchgeo,AugmentationSequential:
```
self.test_transform = AugmentationSequential(
            [K.Normalize(mean=0.0, std=255.0), PadToMultiple(32)], data_keys=["image", "mask"]
        )
```
It complains because even though, `PadToMultiple()` is a `nn.Module` but contains a `Kornia.PadTo` inside:
`NotImplementedError: Only nn.Module are supported at this moment. Got [Normalize(Normalize(p=1.0, p_batch=1.0, same_on_batch=True, mean=tensor([0.]), std=tensor([255.]))), PadToMultiple()].`

Reply:
I have also tried the following:
```
self.test_transform = Compose(
            [
                K.AugmentationSequential(
                    K.Normalize(mean=0.0, std=255.0),
                    data_keys=["input", "mask"]
                ), 
                PadToMultiple(32)], 
        )
```
even though, you already mentioned that this is a bad idea. It also doesn't work because Composes passes a dict,
while Kornia internally expects a tuple of the dict values, but instead gets `({"image":.., "key":,,}, )`and then there is an error because `len(args) != len(data_keys)` since `len(args)=1`

Reply:
For training transforms I don't believe we need to pad since we are random cropping. For test and val sets we do need to do though.

Reply:
new approach to use the `K.Normalize` function with our custom `NCrop` or `PatToMultiple`:
```
self.train_transform = nn.Sequential(
    torchgeo.transforms.AugmentationSequential(
            K.Normalize(mean=0., std=255.), data_keys=["image", "mask"]
    ),
    NCrop(patch_size, self.num_patches_per_tile)
 )
```
This works for setting up the augmentation, however, Kornia always expects floats and currently the `__getitem__` from the deepGlobe dataset returns a `torch.uint8` so should it be changed there?

Reply:
Currently, the last commit includes a version, where the transforms are chained with `torch.nn.Sequential`. The wrapper `torchgeo.transforms.AugmentationSequential` uses `K.AugmentationSequential` internally, and I get errors when trying to use that, becuase internally, the sample dict is not passed from module to module I think but processes the dict items separately as tuples. I am not really familiar with Kornia, so maybe @isaaccorley can comment on that?


```
from torchgeo.transforms import AugmentationSequential, NCrop
import torch
import kornia.augmentation as K

image = torch.rand((3, 256, 256))
mask = torch.randint(high=10, size=(256, 256))
sample = {"image": image, "mask": mask}

# works  (saw this being done in tests/transforms/test_transforms.py
transforms = torch.nn.Sequential(
       AugmentationSequential(
            K.Normalize(mean=0.0, std=255.0), data_keys=["image", "mask"]
        ),
        NCrop(patch_size, self.num_patches_per_tile),
)

# does not work
transforms = AugmentationSequential(
    K.Normalize(mean=0., std=255.), NCrop((32, 32), 3), data_keys=["image", "mask"]
)
```